rs/execution_environment/src/scheduler/tests.rs

```diff
@@ -5815,3 +5823,86 @@ fn scheduled_heap_delta_limit_scaling() {
     assert_eq!(10, scheduled_limit(50, 50, 9, 10, 5));
     assert_eq!(10, scheduled_limit(55, 50, 9, 10, 5));
 }
+
+#[test_strategy::proptest(ProptestConfig { cases: 8, ..ProptestConfig::default() })]
+fn apply_scheduling_strategy_in_inner_round(#[strategy(2..10_usize)] scheduler_cores: usize) {
```

Comment:
This is the change in behavior: we charge all canisters for inner round execution, not only first `scheduler_cores` canisters at the beginning of the round.

Reply:
Just to clarify: we still only charge the first 4 canisters in each loop, not all the canisters that got a (chance at a) "full round", right? Or am I missing anything?

Reply:
Sure, I've updated the comment.